ls8/cpu.c

```diff
@@ -121,19 +131,29 @@ void cpu_run(struct cpu *cpu)
     {
     case HLT:
       running = 0;
+      printf("END PROGRAM\n");
       break;
 
     case PRN:
       // unsigned int value = cpu->ram[cpu->pc++];
-      printf("\"%u\"", cpu->ram[cpu->pc++]);
+      printf("\"%u\"", cpu->reg[cpu->pc]);
+      cpu->pc++;
       break;
 
     case LDI:
       /* LDI reg int */
-      cpu->reg[cpu->pc + 1] = cpu->ram[cpu->pc + 2];
+      printf("load %u into reg %u\n", cpu->ram[cpu->pc + 1], cpu->reg[cpu->pc]);
+      cpu->reg[cpu->pc] = cpu->ram[cpu->pc + 1];
```

Comment:
I notice you commented it out down there, but I was going to suggest: storing stuff in variables can make it quite a bit easier for you (and others) to debug later.